src/main_gate.rs

```diff
@@ -518,8 +518,10 @@ impl<F: PrimeField, const T: usize> MainGate<F, T> {
                 ctx.assign_fixed(|| "q_1", self.config.q_1[i], *val)?;
             }
         }
-        if let Some(q_m_val) = state.1 {
-            ctx.assign_fixed(|| "q_m", self.config.q_m[0], q_m_val)?;
+        if let Some(q_m) = state.1 {
+            for (i, val) in q_m.iter().enumerate() {
+                ctx.assign_fixed(|| "q_m", self.config.q_m[i], *val)?;
+            }
         }
```

Comment:
Size check needed:
```rust
if let Some(q_m) = state.1 {
    q_m.iter()
        .zip_eq(self.config.q_m.iter())
        .try_for_each(|(val, column)| {
            ctx.assign_fixed(|| "q_m", *column, *val)?;
            Result::<_, Error>::Ok(())
        })?;
}
```
or
```rust
if let Some(q_m) = state.1 {
    q_m.iter()
        .zip_eq(self.config.q_m.iter())
        .try_for_each(|(val, column)| {
            ctx.assign_fixed(|| "q_m", *column, *val).map(|_| ())
        })?;
}
```

Reply:
We can't use `zip_eq` here because the length of `q_m` might be less than `self.config.q_m`. I slightly modify your code to use `zip` instead.